crates/bevy_ecs/src/system/function_system.rs

```diff
@@ -286,9 +367,9 @@ macro_rules! impl_system_function {
                 FnMut(In<Input>, $(<<$param as SystemParam>::Fetch as SystemParamFetch>::Item),*) -> Out + Send + Sync + 'static, Out: 'static
         {
             #[inline]
-            fn run(&mut self, input: Input, state: &mut <($($param,)*) as SystemParam>::Fetch, system_state: &SystemState, world: &World, change_tick: u32) -> Out {
+            fn run(&mut self, input: Input, state: &mut <($($param,)*) as SystemParam>::Fetch, system_meta: &SystemMeta, world: &World, change_tick: u32) -> Out {
                 unsafe {
```

Comment:
While the change set doesn't apply to the unsafe block, a `SAFE:` explanation would be nice :)

Reply:
Lol funny story: this isn't safe. This isn't a major crisis as nothing relies on this being safe, but this function needs to be marked as unsafe / needs some safety documentation on the trait.